vali_objects/position.py

```diff
@@ -175,19 +175,33 @@ def _handle_liquidation(self, time_ms):
 
         self.close_out_position(time_ms)
 
-    def set_returns(self, realtime_price, net_leverage, time_ms=None):
+    def calculate_return_with_fees(self, current_return_no_fees, net_leverage, trade_pair, time_ms):
+        # Legacy fee calculation
+        if time_ms < 1711496820246:  # TimeUtil.now_in_millis() as of this update
+            fee = .003 if trade_pair.is_crypto else self.trade_pair.fees
+        else:
+            # V2 fee calculation. Crypto fee lowered from .003 to .002. Multiply fee by leverage for crypto pairs.
+            if trade_pair.is_crypto:
+                fee = self.trade_pair.fees * net_leverage
+            else:
+                fee = self.trade_pair.fees
```

Comment:
why don't we want leverage fees on traditional trades again?

Reply:
Arrash says that this behavior mirrors the real world